LicenseHeaderManager.Console/Program.cs

```diff
@@ -0,0 +1,229 @@
+﻿using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Core;
+using Core.Options;
+
+namespace LicenseHeaderManager.Console
+{
+  public static class Program
+  {
+    /// <summary>
+    /// Specifies if the license headers should be inserted or removed from the files.
+    /// </summary>
+    private enum UpdateMode
+    {
+      Add = 0,
+      Remove = 1
+    }
+
+    public static int Main(string[] args)
+    {
+      var rootCommand = new RootCommand
+                        {
+                            new Option<string> (
+                                new[] { "-m", "--mode" },
+                                () => "add",
+                                "Specifies whether license headers should be added or removed. Must be \"add\" or \"remove\" (case-insensitive)."),
+                            new Option<FileInfo> (
+                                new[] { "-c", "--configuration" },
+                                () => null,
+                                "Specifies the path to the license header definition file to be used for the update operations. Must be present."),
+                            new Option<FileInfo[]> (
+                                new[] { "-f", "--files" },
+                                () => null,
+                                "Specifies the path (or paths) to the files whose headers should be updated. Must not be present if \"directory\" is present."),
+                            new Option<DirectoryInfo> (
+                                new[] { "-d", "--directory" },
+                                () => null,
+                                "Specifies the path of the directory containing the files whose headers should be updated. Must not be present if \"files\" is present."),
+                            new Option<bool?> (
+                                new[] { "-r", "--recursive" },
```

Comment:
I would suggest doing these options a little differently to be more aligned with how other CLI tools do it: I would suggest that we don't require explicitly using this flag if --directory is used. Instead, if not further specified, the search should **not** be recursive. If the user wants it to be recursive, they should add the flag. Simply having `-r` or `--recursive` in the argument string should be enough to trigger the behavior, these flags shouldn't have to be used like `-r true`. Furthermore, even when passing `true` or `false` it doesn't seem to work, I get the following output:

```
C:\Development\LicenseHeaderManager\LicenseHeaderManager.Console\bin\Debug>LicenseHeaderManager.Console.exe -d C:\Development\LicenseHeaderManager -r true

Encountered an unhandled error:
System.NullReferenceException: Object reference not set to an instance of an object.
   at LicenseHeaderManager.Console.Program.UpdateLicenseHeadersForDirectory(UpdateMode mode, FileSystemInfo headerDefinitionFile, DirectoryInfo directory, Boolean recursive) in C:\Development\LicenseHeaderManager\LicenseHeaderManager.Console\Program.cs:line 147
   at LicenseHeaderManager.Console.Program.UpdateLicenseHeaders(UpdateMode mode, FileSystemInfo configuration, IReadOnlyList`1 files, DirectoryInfo directory, Nullable`1 recursive) in C:\Development\LicenseHeaderManager\LicenseHeaderManager.Console\Program.cs:line 116
   at LicenseHeaderManager.Console.Program.<>c.<Main>b__1_0(String mode, FileInfo configuration, FileInfo[] files, DirectoryInfo directory, Nullable`1 recursive) in C:\Development\LicenseHeaderManager\LicenseHeaderManager.Console\Program.cs:line 80
```

So either there's a bug or I can't seem to get the usage right, which is a usability problem. (BTW I just found out the bug was that I didn't specify the .licenseheader file and the application didn't prompt me to, so please add a case for that as well)

Lastly, this is more of a personal preference thing, but I would love to have some colors, at least when printing the result, green for success and red for failure, I don't think it's too much work and adds a little bit of niceness to the application in my opinion.

I'm also not saying that @raffaelfoidl should be the only one to add the requested features to this and the other PR, I'm perfectly willing to do these things myself if I have the time, we just have to coordinate a little bit